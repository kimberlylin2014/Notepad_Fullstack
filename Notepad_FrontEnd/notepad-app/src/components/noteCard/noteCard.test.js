import React from 'react';
import { shallow } from 'enzyme';
import NoteCard from './noteCard.component';

it('Test if Note Card component is rendering', () => {
    const mockPostData = {id: 207, post: "zxc", created: "Sun 9/6/20 5:24pm", modified: false}
    const wrapper = shallow(<NoteCard postData={mockPostData}/>);
    expect(wrapper).toMatchSnapshot()
})


it('Test if Note Card modified is true', () => {
    const mockPostData = {id: 207, post: "zxc", created: "Sun 9/6/20 5:24pm", modified: true}
    const wrapper = shallow(<NoteCard postData={mockPostData}/>);
    expect(wrapper.find('[className="modified"]').text()).toEqual(`Modified`)
})

it('Test if Note Card modified is false', () => {
    const mockPostData = {id: 207, post: "zxc", created: "Sun 9/6/20 5:24pm", modified: false}
    const wrapper = shallow(<NoteCard postData={mockPostData}/>);
    expect(wrapper.find('[className="created"]').text()).toEqual(`Created:`)
})

it('Test Note Card Click Event', () => {
    const mockPostData = {id: 207, post: "zxc", created: "Sun 9/6/20 5:24pm", modified: false}
    const mockFunction = jest.fn()
    const wrapper = shallow(<NoteCard postData={mockPostData} deletePostStart={mockFunction}/>);
    wrapper.find('[className="delete-icon"]').simulate('click');
    expect(mockFunction.mock.calls.length).toBe(1)
})