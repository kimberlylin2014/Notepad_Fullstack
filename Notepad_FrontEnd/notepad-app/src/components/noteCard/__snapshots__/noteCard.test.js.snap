// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Note Card component is rendering 1`] = `
<div
  className="NoteCard"
>
  <div
    className="header"
  >
    <EditNoteModal
      buttonLabel="Edit"
      postData={
        Object {
          "created": "Sun 9/6/20 5:24pm",
          "id": 207,
          "modified": false,
          "post": "zxc",
        }
      }
    />
    <div
      className="delete-icon"
      onClick={[Function]}
    >
      ✖
    </div>
  </div>
  <div
    className="text-section"
  >
    <p
      className="note-text"
    >
      zxc
    </p>
    <p
      className="timestamp"
    >
       
      <span
        className="created"
      >
        Created:
      </span>
        
      Sun 9/6/20 5:24pm
    </p>
  </div>
</div>
`;
